### PR TITLE
add all event attribute types

### PR DESCRIPTION
**Pull request by HashZhang**
- add all event attribute types
 - add predicate about class and thread

### CODE REVIEW

File: src/test/java/dev/morling/jfrunit/JfrUnitTest.java [hunk 1]
```diff
@@ -70,4 +73,48 @@ public void shouldHaveGcAndSleepEventsWithDefaultConfiguration() throws Exceptio
...
+                .with("allocationSize", (double) OBJECT_SIZE)
+                .with("objectClass", new ExpectedClass(new byte[0].getClass()))
+                .with("eventThread", new ExpectedThread(Thread.currentThread()))
+        ).collect(Collectors.toList());
+        List<RecordedEvent> allocation100KBOutsideTLABEvents2 = jfrEvents.filter(event("jdk.ObjectAllocationOutsideTLAB")
+                .with("allocationSize", (double) OBJECT_SIZE)
+                .with("objectClass", new ExpectedClass(new byte[0].getClass()))
+                .with("eventThread", new ExpectedThread(Thread.currentThread()))
+        ).collect(Collectors.toList());
+        assertThat(allocation100KBInNewTLABEvents.size()).isEqualTo(allocation100KBInNewTLABEvents2.size());
```
Comment (gunnarmorling) [inline src/test/java/dev/morling/jfrunit/JfrUnitTest.java:115]: What's the purpose of this assertion? I.e. why do it again and compare?

File: src/test/java/dev/morling/jfrunit/JfrUnitTest.java [hunk 2]
```diff
@@ -70,4 +73,48 @@ public void shouldHaveGcAndSleepEventsWithDefaultConfiguration() throws Exceptio
...
+        assertThat(allocation100KBOutsideTLABEvents.size()).isEqualTo(allocation100KBOutsideTLABEvents2.size());
+
+        //todo add more tlab test
```
Comment (gunnarmorling) [inline src/test/java/dev/morling/jfrunit/JfrUnitTest.java:118]: Could you add tests for all the things you added, like thread group, method, etc.?

File: src/test/java/dev/morling/jfrunit/JfrUnitTest.java [hunk 3]
```diff
@@ -70,4 +73,48 @@ public void shouldHaveGcAndSleepEventsWithDefaultConfiguration() throws Exceptio
 
         assertThat(allocated).isGreaterThan(0);
     }
+
+    private static final int BYTE_ARRAY_OVERHEAD = 16;
+    private static final int OBJECT_SIZE = 102400;
+    private static final String BYTE_ARRAY_CLASS_NAME = new byte[0].getClass().getName();
```
Comment (gunnarmorling) [inline src/test/java/dev/morling/jfrunit/JfrUnitTest.java:79]:
```suggestion
    private static final String BYTE_ARRAY_CLASS_NAME = byte[].class.getName();
```

**Review by gunnarmorling — COMMENTED** [review]
Thanks a lot, this LGTM overall. A few small comments/suggestions inline.

File: src/test/java/dev/morling/jfrunit/JfrUnitTest.java (hunk 1 above)
Comment (HashZhang) [reply src/test/java/dev/morling/jfrunit/JfrUnitTest.java]: updated

File: src/test/java/dev/morling/jfrunit/JfrUnitTest.java (hunk 2 above)
Comment (HashZhang) [reply src/test/java/dev/morling/jfrunit/JfrUnitTest.java]: updated

**Review by gunnarmorling — APPROVED** [review]
LGTM, thanks a lot!

**Comment by gunnarmorling** [comment]
Squashed and applied. Thanks a lot, @HashZhang!

**Pull request closed without merging**